elasticdl_preprocessing/tests/hashing_test.py

```diff
@@ -0,0 +1,31 @@
+import unittest
+
+import numpy as np
+import tensorflow as tf
+
+from elasticdl_preprocessing.layers.hashing import Hashing
+from elasticdl_preprocessing.tests.test_utils import (
+    ragged_tensor_equal,
+    sparse_tensor_equal,
+)
+
+
+class HashingTest(unittest.TestCase):
+    def test_hashing(self):
+        hash_layer = Hashing(num_bins=3)
+        inp = np.asarray([["A"], ["B"], ["C"], ["D"], ["E"]])
+        hash_out = hash_layer(inp)
+        expected_out = np.array([[1], [0], [1], [1], [2]])
+        self.assertTrue(np.array_equal(hash_out.numpy(), expected_out))
+
+        ragged_in = tf.ragged.constant([["A", "B"], ["C", "D"], ["E"], []])
+        hash_out = hash_layer(ragged_in)
+        expected_ragged_out = tf.ragged.constant(
+            [[1, 0], [1, 1], [2], []], dtype=tf.int64
+        )
+        self.assertTrue(ragged_tensor_equal(hash_out, expected_ragged_out))
+
+        sparse_in = ragged_in.to_sparse()
+        hash_out = hash_layer(sparse_in)
+        expected_sparse_out = expected_ragged_out.to_sparse()
+        self.assertTrue(sparse_tensor_equal(hash_out, expected_sparse_out))
```

Comment:
Check hash_out to be SparseTensor?

Reply:
Yes, `sparse_tensor_equal` will check it. See the codes in #1846
https://github.com/sql-machine-learning/elasticdl/blob/e580f27bb95ab7b998ec436a4f688fafcea2d7c6/elasticdl_preprocessing/tests/test_utils.py#L5-L8